Add App Modernization related documentation for VS Code Java node (#8901)
microsoft/vscode-docs: docs/java/java-app-mod.md

# Modernizing your legacy Java Apps

The [GitHub Copilot App Modernization extension](https://marketplace.visualstudio.com/items?itemName=vscjava.migrate-java-to-azure) (part of the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack)) provides powerful capabilities to help developers modernize legacy Java applications with greater speed, accuracy, and efficiency.
Whether upgrading frameworks, addressing outdated dependencies, or preparing for cloud migration, GitHub Copilot leverages intelligent automation and deep integration with Microsoft Azure to accelerate modernization—from months down to days

![GitHub Copilot App Modernization](images/java-app-mod/appmodlogo.png)

The extension offer the following the key capabilities:

> 🧠 Intelligent assessments and upgrade planning

Reviewer: ```suggestion
* Intelligent assessments and upgrade planning
```

Copilot scans your codebase, analyzes dependencies and frameworks, and builds actionable plans to guide your modernization journey.

> 🛠️ AI-powered code transformation

Reviewer: ```suggestion
* AI-powered code transformation
```

> From Java 8 to latest LTS version, Copilot automates upgrades with structured, context-aware edits—cutting out repetitive work and reducing upgrade risk.

Reviewer: ```suggestion
    From Java 8 to the latest LTS version, Copilot automates upgrades with structured, context-aware edits,  cutting out repetitive work and reducing upgrade risk.
```

> 🔁 Reusable migration patterns

Reviewer: ```suggestion
* Reusable migration patterns
```

> Capture your team’s refactoring logic into custom formulas, then reapply them across apps for consistent, scalable modernization.

Reviewer: ```suggestion
    Capture your team’s refactoring logic into custom formulas, then reapply them across apps for consistent, scalable modernization.
```

> ✅ Build and test validation

Reviewer: ```suggestion
* Build and test validation
```

Copilot fixes build errors, runs unit tests, and ensures your application is stable after changes—helping maintain clean CI/CD pipelines.

> 🛡️ CVE detection and remediation

Reviewer: ```suggestion
* CVE detection and remediation
```

Automatically detect and patch known vulnerabilities post-upgrade. Copilot applies secure fixes while keeping developers in full control of the changes.

> ☁️ Seamless Azure integration

Reviewer: ```suggestion
* Seamless Azure integration
```

> Deploy upgraded apps to Azure with minimal effort. Modernized apps can immediately tap into Azure AI Foundry to unlock 11,000+ models, built-in observability, safety tools, and agent services.

Reviewer: ```suggestion
    Deploy upgraded apps to Azure with minimal effort. Modernized apps can immediately use Azure AI Foundry to unlock over 11,000 models, built-in observability, safety tools, and agent services.
```
